src/resmom/catch_child.c

```diff
@@ -1846,79 +1813,71 @@ scan_for_exiting(void)
 
 		write_wkmg_record(WM_TERM, WM_TERM_EXIT, pjob);
 #endif	/* MOM_CSA */
-
-		if (mock_run) {
+#endif
+		if (mock_run || !has_epilog){
 			send_obit(pjob, 0);
 			continue;
 		}
 
+#ifndef WIN32
 		/*
 		 * Parent:
 		 *  +  fork child process to run epilogue,
 		 *  +  look for more terminated jobs.
 		 * Child:
-		 *  +  Run the epilogue script (if one)
+		 *  +  Run the epilogue script
 		 */
 
 		cpid = fork_me(-1);
 		if (cpid > 0) {
-			/* parent = mark that it is being sent */
-			pjob->ji_sampletim = 0;	/* will be set in send_obit */
+			pjob->ji_sampletim = 0;
 			pjob->ji_momsubt = cpid;
 			pjob->ji_actalarm = 0;
 			pjob->ji_mompost = send_obit;
 			pjob->ji_qs.ji_substate = JOB_SUBSTATE_RUNEPILOG;
 
-			if (found_one++ < 5) {
-				continue;	/* look for more */
+			if (found_one++ < 20) {
+				continue; /* look for more exiting jobs */
 			} else {
-				break;	/* five at a time is our limit */
+				break; /* 20 exiting jobs at a time is our limit */
 			}
 		} else if (cpid < 0)
-			continue;	/* curses, foiled again */
-
-		/* child: change to the user's home directory or PBS_JOBDIR */
-		/* and run the epilogue script				    */
+			continue; /* curses, failed again */
+#endif
 
 		if (pjob->ji_grpcache) {
 			if ((pjob->ji_wattr[(int)JOB_ATR_sandbox].at_flags & ATR_VFLAG_SET) && (strcasecmp(pjob->ji_wattr[JOB_ATR_sandbox].at_val.at_str, "PRIVATE") == 0)) {
-				/* in "sandbox=PRIVATE" mode so run epilogue */
-				/* in PBS_JOBDIR */
-				(void)chdir(jobdirname(pjob->ji_qs.ji_jobid,
-					pjob->ji_grpcache->gc_homedir));
-
+				/* in "sandbox=PRIVATE" mode so run epilogue in PBS_JOBDIR */
+				(void)chdir(jobdirname(pjob->ji_qs.ji_jobid, pjob->ji_grpcache->gc_homedir));
 			} else {
-
 				/* else run in usr's home */
 				(void)chdir(pjob->ji_grpcache->gc_homedir);
 			}
 		}
 
 		extval = 0;
-
 		if (num_eligible_hooks(HOOK_EVENT_EXECJOB_EPILOGUE) > 0) {
 			mom_hook_input_init(&hook_input);
 			hook_input.pjob = pjob;
-			(void)mom_process_hooks(HOOK_EVENT_EXECJOB_EPILOGUE,
-				PBS_MOM_SERVICE_NAME, mom_host, &hook_input,
-				NULL, NULL, 0, 0);
+			(void)mom_process_hooks(HOOK_EVENT_EXECJOB_EPILOGUE, PBS_MOM_SERVICE_NAME, mom_host, &hook_input, NULL, NULL, 0, 0);
```

Comment:
I see that the #ifndef WIN32 has been removed so that the same mom_process_hooks() call is done. But there's a subtle difference. The last parameter is actually a '1' (update_server) on the Windows side since it is not executing under forked process. It should be:
if (mom_process_hooks(HOOK_EVENT_EXECJOB_EPILOGUE,
                        PBS_MOM_SERVICE_NAME, mom_host, &hook_input, NULL,
                        NULL, 0, **1**)

Reply:
oops, nice catch, missed that. will change it.